In [1]:
import numpy as np
import pandas as pd 
import os
import matplotlib.pyplot as plt
import seaborn as sns
for dirname, _, filenames in os.walk('./kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

./kaggle/input/broad-peak.csv
./kaggle/input/lhotse.csv
./kaggle/input/kangchenjunga.csv
./kaggle/input/cho-oyu.csv
./kaggle/input/manaslu.csv
./kaggle/input/dhaulagiri-I.csv
./kaggle/input/everest.csv
./kaggle/input/k2.csv
./kaggle/input/nanga-parbat.csv
./kaggle/input/gasherbrum-II.csv
./kaggle/input/shishapangma.csv
./kaggle/input/gasherbrum-I.csv
./kaggle/input/annapurna-I.csv
./kaggle/input/makalu.csv


In [2]:
broadpeek = pd.read_csv('./kaggle/input/broad-peak.csv')
lhotse = pd.read_csv('./kaggle/input/lhotse.csv')
kangchenjunga = pd.read_csv('./kaggle/input/kangchenjunga.csv')
chooyu = pd.read_csv('./kaggle/input/cho-oyu.csv')
manaslu = pd.read_csv('./kaggle/input/manaslu.csv')
dhaulagiri = pd.read_csv('./kaggle/input/dhaulagiri-I.csv')
everest = pd.read_csv('./kaggle/input/everest.csv')
k2 = pd.read_csv('./kaggle/input/k2.csv')
nangaparbat = pd.read_csv('./kaggle/input/nanga-parbat.csv')
gasherbrum2 = pd.read_csv('./kaggle/input/gasherbrum-II.csv')
gasherbrum1 = pd.read_csv('./kaggle/input/gasherbrum-I.csv')
shishapangma = pd.read_csv('./kaggle/input/shishapangma.csv')
annapurna1 = pd.read_csv('./kaggle/input/annapurna-I.csv')
makalu = pd.read_csv('./kaggle/input/makalu.csv')

In [3]:
# List number of rows
everest.shape

(311, 6)

In [4]:
# Assign mountain in each dataset
broadpeek['mountain'] = 'Broad Peak'
lhotse['mountain'] = 'Lhotse'
kangchenjunga['mountain'] = 'Kangchenjunga'
chooyu['mountain'] = 'Chooyu'
manaslu['mountain'] = 'Manaslu'
dhaulagiri['mountain'] = 'Dhaulagiri I'
everest['mountain'] = 'Everest'
k2['mountain'] = 'K2'
nangaparbat['mountain'] = 'Nanga Parbat'
gasherbrum1['mountain'] = 'Gasherbrum I'
gasherbrum2['mountain'] = 'Gasherbrum II'
makalu['mountain'] = 'Makalu'
annapurna1['mountain'] = 'Annapurna I'
shishapangma['mountain'] = 'Shishapangma'

# Join datasets
allmountains = pd.concat([
    broadpeek, lhotse, kangchenjunga, chooyu, manaslu, dhaulagiri,
    everest, k2, nangaparbat, gasherbrum2, gasherbrum1, 
    shishapangma, annapurna1, makalu
], ignore_index=True)

In [5]:
allmountains

,Date,Name,Nationality,Cause of death,"Lat, Lon",Rank,mountain
0,7/19/2021,Kim Hong-bin,South Korea,Fall,"35.8107° N, 76.5680° E",12,Broad Peak
1,7/1/2015,Qamber Ali Jangjupa,Pakistan,Avalanche,"35.8107° N, 76.5680° E",12,Broad Peak
2,7/23/2013,Aidin Bozorgi,Iran,Disappeared,"35.8107° N, 76.5680° E",12,Broad Peak
3,7/23/2013,Mojtaba Jarrahi [fa],Iran,Disappeared,"35.8107° N, 76.5680° E",12,Broad Peak
4,7/23/2013,Pouya Keivan [fa],Iran,Disappeared,"35.8107° N, 76.5680° E",12,Broad Peak
...,...,...,...,...,...,...,...
1028,9/26/1982,Tadeusz Szulc,Poland,Illness,"27.7400° N, 87.1500° E",5,Makalu
1029,10/6/1978,Andrzej Młynarczyk,Poland,Avalanche,"27.7400° N, 87.1500° E",5,Makalu
1030,5/25/1976,Karel Schubert,Czechoslovakia,Exhaustion,"27.7400° N, 87.1500° E",5,Makalu
1031,5/26/1973,Jan Kounický,Czechoslovakia,Fall,"27.7400° N, 87.1500° E",5,Makalu


In [6]:
# Deaths per mountain
allmountains['mountain'].value_counts()

mountain
Everest          311
K2                92
Manaslu           86
Nanga Parbat      85
Dhaulagiri I      83
Annapurna I       73
Kangchenjunga     57
Chooyu            52
Makalu            40
Broad Peak        35
Gasherbrum I      34
Lhotse            31
Shishapangma      31
Gasherbrum II     23
Name: count, dtype: int64

In [7]:
# Deaths per nationality
allmountains['Nationality'].value_counts()

Nationality
Nepal          220
Japan           89
South Korea     53
Spain           51
India           48
              ... 
?                1
Macedonia        1
Belarus          1
Hong Kong        1
Peru             1
Name: count, Length: 69, dtype: int64

In [8]:
# Deaths by cause
allmountains['Cause of death'].value_counts()

Cause of death
Avalanche                284
Fall                     218
Disappeared              100
Altitude sickness         66
Exhaustion                41
                        ... 
Snowboarding accident      1
Coma                       1
Rope accident              1
Injuries from a fall       1
Rockfall                   1
Name: count, Length: 80, dtype: int64

In [9]:
# Format dates
allmountains['Date'] = pd.to_datetime(allmountains['Date'], format='mixed', errors='coerce')
allmountains['year'] = allmountains['Date'].dt.year
allmountains['month'] = allmountains['Date'].dt.month

# Check if any dates failed to parse
print(f"Missing dates: {allmountains['Date'].isna().sum()}")

# Deaths per year
allmountains['year'].value_counts().sort_index()

Missing dates: 0


year
1895     3
1905     3
1922     7
1924     4
1929     1
        ..
2017     9
2018    16
2019    25
2021    12
2022     4
Name: count, Length: 76, dtype: int64

In [10]:
# What % of deaths does each mountain account for?
(allmountains['mountain'].value_counts(normalize=True) * 100).round(2)

mountain
Everest          30.11
K2                8.91
Manaslu           8.33
Nanga Parbat      8.23
Dhaulagiri I      8.03
Annapurna I       7.07
Kangchenjunga     5.52
Chooyu            5.03
Makalu            3.87
Broad Peak        3.39
Gasherbrum I      3.29
Lhotse            3.00
Shishapangma      3.00
Gasherbrum II     2.23
Name: proportion, dtype: float64

In [11]:
# Top 10 nationalities by death count
allmountains['Nationality'].value_counts().head(10)

Nationality
Nepal             220
Japan              89
South Korea        53
Spain              51
India              48
France             45
United States      43
United Kingdom     36
Germany            34
Poland             34
Name: count, dtype: int64

In [12]:
# Month when most deaths occur
allmountains['month'] = pd.to_datetime(allmountains['Date']).dt.month
allmountains['month'].value_counts().sort_index()

month
1      11
2       8
3      18
4     136
5     307
6      79
7     124
8      69
9     116
10    140
11      1
12     24
Name: count, dtype: int64

In [13]:
# Deaths per decade
allmountains['decade'] = (allmountains['year'] // 10) * 10
allmountains['decade'].value_counts().sort_index()

decade
1890      3
1900      3
1920     12
1930     36
1950     15
1960     16
1970     99
1980    206
1990    216
2000    174
2010    237
2020     16
Name: count, dtype: int64

In [14]:
# Nationality vs mountain
pd.crosstab(allmountains['Nationality'], allmountains['mountain'])

mountain,Annapurna I,Broad Peak,Chooyu,Dhaulagiri I,Everest,Gasherbrum I,Gasherbrum II,K2,Kangchenjunga,Lhotse,Makalu,Manaslu,Nanga Parbat,Shishapangma
Nationality,,,,,,,,,,,,,,
?,0,0,0,0,1,0,0,0,0,0,0,0,0,0
Argentina,0,0,0,3,0,1,0,0,0,0,0,1,1,0
Australia,0,0,1,1,7,0,0,0,0,0,3,1,0,1
Austria,1,1,0,2,3,1,2,3,0,0,0,5,3,0
Bangladesh,0,0,0,0,1,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
Uzbekistan,2,0,0,0,0,0,0,0,0,0,0,0,0,0
Venezuela,0,0,0,0,0,0,0,0,0,0,0,0,1,0
Vietnam,0,0,0,0,1,0,0,0,0,0,0,0,0,0


In [15]:
# Date range of data
print(f"From {allmountains['Date'].min()} to {allmountains['Date'].max()}")

From 1895-08-24 00:00:00 to 2022-05-08 00:00:00


In [16]:
# Number of unique nationalities represented
allmountains['Nationality'].nunique()

69

In [17]:
# Bottom 10 nationalities by death count
allmountains['Nationality'].value_counts().tail(10)

Nationality
Iceland       1
Bolivia       1
Brazil        1
Bangladesh    1
Vietnam       1
?             1
Macedonia     1
Belarus       1
Hong Kong     1
Peru          1
Name: count, dtype: int64

In [18]:
# Top 10 nationalities by death count
allmountains['Nationality'].value_counts().head(10)

Nationality
Nepal             220
Japan              89
South Korea        53
Spain              51
India              48
France             45
United States      43
United Kingdom     36
Germany            34
Poland             34
Name: count, dtype: int64

In [19]:
# Top 10 mountains by death count
allmountains['mountain'].value_counts().head(10)

mountain
Everest          311
K2                92
Manaslu           86
Nanga Parbat      85
Dhaulagiri I      83
Annapurna I       73
Kangchenjunga     57
Chooyu            52
Makalu            40
Broad Peak        35
Name: count, dtype: int64

In [20]:
# Create a crosstab of deaths by year and mountain
deaths_by_year_mountain = pd.crosstab(allmountains['year'], allmountains['mountain'])

# Display the table
print("DEATHS BY YEAR AND MOUNTAIN")
print("=" * 150)
print(deaths_by_year_mountain)

# If you want to save it to CSV
deaths_by_year_mountain.to_csv('deaths_by_year_mountain.csv')

# Summary statistics
print("\n" + "=" * 150)
print("SUMMARY STATISTICS")
print("=" * 150)
print(f"Total years covered: {len(deaths_by_year_mountain)}")
print(f"Year range: {deaths_by_year_mountain.index.min()} to {deaths_by_year_mountain.index.max()}")
print(f"\nTotal deaths per mountain:")
print(deaths_by_year_mountain.sum().sort_values(ascending=False))

DEATHS BY YEAR AND MOUNTAIN
mountain  Annapurna I  Broad Peak  Chooyu  Dhaulagiri I  Everest  \
year                                                               
1895                0           0       0             0        0   
1905                0           0       0             0        0   
1922                0           0       0             0        7   
1924                0           0       0             0        4   
1929                0           0       0             0        0   
...               ...         ...     ...           ...      ...   
2017                0           0       0             1        5   
2018                0           0       1             2        5   
2019                1           0       1             0       12   
2021                0           1       0             0        4   
2022                0           0       0             1        2   

mountain  Gasherbrum I  Gasherbrum II  K2  Kangchenjunga  Lhotse  Makalu  \
year       

In [21]:
# Show only non-zero values (years where deaths occurred)
deaths_table = pd.crosstab(allmountains['year'], allmountains['mountain'])
deaths_table = deaths_table.loc[(deaths_table != 0).any(axis=1)]  # Remove years with no deaths

print(deaths_table)

mountain  Annapurna I  Broad Peak  Chooyu  Dhaulagiri I  Everest  \
year                                                               
1895                0           0       0             0        0   
1905                0           0       0             0        0   
1922                0           0       0             0        7   
1924                0           0       0             0        4   
1929                0           0       0             0        0   
...               ...         ...     ...           ...      ...   
2017                0           0       0             1        5   
2018                0           0       1             2        5   
2019                1           0       1             0       12   
2021                0           1       0             0        4   
2022                0           0       0             1        2   

mountain  Gasherbrum I  Gasherbrum II  K2  Kangchenjunga  Lhotse  Makalu  \
year                                   

In [22]:
# Styled version with totals
deaths_table = pd.crosstab(allmountains['year'], allmountains['mountain'], margins=True)
print(deaths_table)

mountain  Annapurna I  Broad Peak  Chooyu  Dhaulagiri I  Everest  \
year                                                               
1895                0           0       0             0        0   
1905                0           0       0             0        0   
1922                0           0       0             0        7   
1924                0           0       0             0        4   
1929                0           0       0             0        0   
...               ...         ...     ...           ...      ...   
2018                0           0       1             2        5   
2019                1           0       1             0       12   
2021                0           1       0             0        4   
2022                0           0       0             1        2   
All                73          35      52            83      311   

mountain  Gasherbrum I  Gasherbrum II  K2  Kangchenjunga  Lhotse  Makalu  \
year                                   

Machine Learning

In [26]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder

# Encode categorical variables
le_mountain = LabelEncoder()
le_nationality = LabelEncoder()

allmountains['mountain_encoded'] = le_mountain.fit_transform(allmountains['mountain'])
allmountains['nationality_encoded'] = le_nationality.fit_transform(allmountains['Nationality'])

# Predict cause of death
X = allmountains[['mountain_encoded', 'year', 'month', 'nationality_encoded']].dropna()
y = allmountains.loc[X.index, 'Cause of death']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = RandomForestClassifier(n_estimators=100, random_state=42)
model.fit(X_train, y_train)

# Accuracy
print(f"Model Accuracy: {model.score(X_test, y_test):.2%}")

# Feature importance
feature_names = ['Mountain', 'Year', 'Month', 'Nationality']
importances = model.feature_importances_

print("\nFeature Importance:")
for name, importance in zip(feature_names, importances):
    print(f"{name}: {importance:.4f}")

Model Accuracy: 53.14%

Feature Importance:
Mountain: 0.1940
Year: 0.3773
Month: 0.1557
Nationality: 0.2731


This machine learning model has an accuracy of 53.14%. This is modest but better than random guessing. Demonstrates that deaths are somewhat predictable based on mountain, year, and nationality.
Year (37.73%): Most important - cause of death patterns change over time.
Nationality (27.31%): Second most - different climbing cultures/experience levels.
Mountain (19.40%): Third - each mountain has different hazards.
Month (15.57%): Least important - season matters less, probably due to mountains have different climbing seasons.

In [27]:
from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.naive_bayes import GaussianNB
from sklearn.svm import SVC
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.preprocessing import LabelEncoder
import pandas as pd

# Prepare data (same as before)
le_mountain = LabelEncoder()
le_nationality = LabelEncoder()

allmountains['mountain_encoded'] = le_mountain.fit_transform(allmountains['mountain'])
allmountains['nationality_encoded'] = le_nationality.fit_transform(allmountains['Nationality'])

X = allmountains[['mountain_encoded', 'year', 'month', 'nationality_encoded']].dropna()
y = allmountains.loc[X.index, 'Cause of death']

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Dictionary of models to test
models = {
    'Random Forest': RandomForestClassifier(n_estimators=100, random_state=42),
    'Gradient Boosting': GradientBoostingClassifier(n_estimators=100, random_state=42),
    'Decision Tree': DecisionTreeClassifier(random_state=42),
    'K-Nearest Neighbors': KNeighborsClassifier(n_neighbors=5),
    'Naive Bayes': GaussianNB(),
    'Support Vector Machine': SVC(random_state=42),
    'Logistic Regression': LogisticRegression(max_iter=1000, random_state=42)
}

# Train and evaluate each model
results = []

print("MACHINE LEARNING MODEL COMPARISON")
print("=" * 70)

for name, model in models.items():
    # Train the model
    model.fit(X_train, y_train)
    
    # Test accuracy
    train_score = model.score(X_train, y_train)
    test_score = model.score(X_test, y_test)
    
    # Cross-validation score (more reliable)
    cv_scores = cross_val_score(model, X, y, cv=5)
    cv_mean = cv_scores.mean()
    
    results.append({
        'Model': name,
        'Train Accuracy': train_score,
        'Test Accuracy': test_score,
        'CV Accuracy': cv_mean
    })
    
    print(f"{name:25s} | Train: {train_score:.2%} | Test: {test_score:.2%} | CV: {cv_mean:.2%}")

print("=" * 70)

# Convert to DataFrame for easy comparison
results_df = pd.DataFrame(results).sort_values('Test Accuracy', ascending=False)

print("\nBEST MODELS (sorted by Test Accuracy):")
print(results_df.to_string(index=False))

# Get the best model
best_model_name = results_df.iloc[0]['Model']
best_model = models[best_model_name]

print(f"\n🏆 WINNER: {best_model_name} with {results_df.iloc[0]['Test Accuracy']:.2%} test accuracy")

# If it's Random Forest or Gradient Boosting, show feature importance
if best_model_name in ['Random Forest', 'Gradient Boosting']:
    feature_names = ['Mountain', 'Year', 'Month', 'Nationality']
    importances = best_model.feature_importances_
    
    print(f"\nFeature Importance for {best_model_name}:")
    for name, importance in sorted(zip(feature_names, importances), key=lambda x: x[1], reverse=True):
        print(f"  {name}: {importance:.4f}")

MACHINE LEARNING MODEL COMPARISON


/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(


Random Forest             | Train: 96.00% | Test: 53.14% | CV: 20.52%


/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(


Gradient Boosting         | Train: 85.84% | Test: 46.86% | CV: 19.65%
Decision Tree             | Train: 96.00% | Test: 45.41% | CV: 18.39%
K-Nearest Neighbors       | Train: 50.36% | Test: 34.30% | CV: 20.33%
Naive Bayes               | Train: 32.45% | Test: 26.09% | CV: 23.34%


/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(


Support Vector Machine    | Train: 27.24% | Test: 28.50% | CV: 27.49%


/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/linear_model/_logistic.py:406: ConvergenceWarning: lbfgs failed to converge after 1000 iteration(s) (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT

Increase the number of iterations to improve the convergence (max_iter=1000).
You might also want to scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/model_selection/_split.py:813: UserWarning: The least populated class in y has only 1 members, which is less than n_splits=5.
  warnings.warn(
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/linear_model/_logistic.py:406: ConvergenceWarning: lbf

Logistic Regression       | Train: 27.60% | Test: 26.57% | CV: 21.97%

BEST MODELS (sorted by Test Accuracy):
                 Model  Train Accuracy  Test Accuracy  CV Accuracy
         Random Forest        0.960048       0.531401     0.205192
     Gradient Boosting        0.858354       0.468599     0.196478
         Decision Tree        0.960048       0.454106     0.183931
   K-Nearest Neighbors        0.503632       0.342995     0.203255
Support Vector Machine        0.272397       0.285024     0.274931
   Logistic Regression        0.276029       0.265700     0.219666
           Naive Bayes        0.324455       0.260870     0.233390

🏆 WINNER: Random Forest with 53.14% test accuracy

Feature Importance for Random Forest:
  Year: 0.3773
  Nationality: 0.2731
  Mountain: 0.1940
  Month: 0.1557


/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/sklearn/linear_model/_logistic.py:406: ConvergenceWarning: lbfgs failed to converge after 1000 iteration(s) (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT

Increase the number of iterations to improve the convergence (max_iter=1000).
You might also want to scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
